frontend-react/src/shared/Tooltip.tsx

```diff
@@ -0,0 +1,215 @@
+import {
+    Placement,
+    useFloating,
+    autoUpdate,
+    flip,
+    shift,
+    useHover,
+    useFocus,
+    useDismiss,
+    useRole,
+    useInteractions,
+    useMergeRefs,
+    FloatingPortal,
+    offset,
+    OffsetOptions,
+} from "@floating-ui/react";
+import classNames from "classnames";
+import React from "react";
+
+export const TOOLTIP_POSITIONS = ["top", "bottom", "right", "left"];
+
+export interface TooltipOptions {
+    initialOpen?: boolean;
+    placement?: Placement;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+    offsetBy?: OffsetOptions;
```

Comment:
We should confirm with Jillian / Audrey about what the default should be here. As the default is way too closely cropped to the element it's stemming from:

<img width="159" alt="Screenshot 2023-04-18 at 12 08 07 PM" src="https://user-images.githubusercontent.com/516877/232880283-3aa8b5b1-95d4-42d2-aeb8-1e438ff085a7.png">

We should also extrapolate for the Developer "approved" offsets in whole numbers like they do in [Trussworks library for Icons](https://github.com/trussworks/react-uswds/blob/main/src/components/Icon/Icon.tsx#L7) (and other custom sized components). So that a developer could pass something like `offset={1}`, `offset={2}`, `offset={3}`, etc. instead of having to pass in a [slew of options](https://floating-ui.com/docs/offset) (since they provide so much like crossAxis, mainAxis etc). I'm hoping our components can be "approved out of the box" instead of having to go to Design with what offsets should be.